app/services/billable_metrics/aggregations/unique_count_service.rb

```diff
@@ -39,6 +51,96 @@ def running_total(options)
 
         (1..result.aggregation).to_a
       end
+
+      private
+
+      attr_reader :from_datetime, :to_datetime
+
+      # This method fetches the latest event in current period. If such a event exists we know that metadata
+      # with previous aggregation and previous maximum aggregation are stored there. Fetching these metadata values
+      # would help us in pay in advance value calculation without iterating through all events in current period
+      def previous_event
+        @previous_event ||= begin
+          query = events_scope(from_datetime:, to_datetime:)
+            .joins(:quantified_event)
+            .where("#{sanitized_field_name} IS NOT NULL")
+            .where.not(id: event.id)
+            .where('quantified_events.added_at::timestamp(0) >= ?', from_datetime)
+            .where('quantified_events.added_at::timestamp(0) <= ?', to_datetime)
+            .order(created_at: :desc)
+
+          query
+            .where('quantified_events.removed_at::timestamp(0) IS NULL')
+            .or(
+              query
+                .where('quantified_events.removed_at::timestamp(0) >= ?', from_datetime)
+                .where('quantified_events.removed_at::timestamp(0) <= ?', to_datetime)
+            )
+
+          query.first
+        end
+      end
+
+      def operation_type
+        @operation_type ||= event.properties.fetch('operation_type', 'add')&.to_sym
+      end
+
+      def handle_event_metadata(current_aggregation: nil, max_aggregation: nil)
+        result.current_aggregation = current_aggregation unless current_aggregation.nil?
+        result.max_aggregation = max_aggregation unless max_aggregation.nil?
+      end
+
+      def compute_aggregation
+        ActiveRecord::Base.connection.execute(aggregation_query).first['aggregation_result']
+      end
+
+      def aggregation_query
+        queries = [
+          # NOTE: Billed on the full period. We will replace 1::numeric with proration_coefficient::numeric
+          # in the next part
+          persisted.select("SUM(1::numeric)").to_sql,
+
+          # NOTE: Added during the period, We will replace 1::numeric with proration_coefficient::numeric
+          # in the next part
+          added.select("SUM(1::numeric)").to_sql,
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://rubystyle.guide#consistent-string-literals)